### PR TITLE
[SYCL][UR] Add missing device queries for bindless images on level zero backend

### PR DESCRIPTION
**Pull request by dyniols**
This PR adds support for pitch alignment restrictions queries in L0 backend. Until now only supported queries were `max_image_linear_width` and `max_image_linear_height`. 

Since L0 added `ze_pitched_alloc_2dimage_linear_pitch_exp_info_t` we can query `image_row_pitch_align` and `max_image_linear_row_pitch`.

https://oneapi-src.github.io/level-zero-spec/level-zero/latest/core/api.html#ze-pitched-alloc-2dimage-linear-pitch-exp-info-t

### CODE REVIEW

File: unified-runtime/source/adapters/level_zero/device.cpp [hunk 1]
```diff
@@ -1265,11 +1265,34 @@ ur_result_t urDeviceGetInfo(
...
+    ze_device_pitched_alloc_exp_properties_t imageAllocProps = {};
+    imageAllocProps.stype =
+        ZE_STRUCTURE_TYPE_PITCHED_ALLOC_DEVICE_EXP_PROPERTIES;
+    ze_pitched_alloc_2dimage_linear_pitch_exp_info_t imageLinearPitchInfo = {};
+    imageLinearPitchInfo.stype = static_cast<ze_structure_type_t>(
+        ZE_STRUCTURE_TYPE_PITCHED_ALLOC_2DIMAGE_LINEAR_PITCH_EXP_INFO);
+    imageProps.pNext = (void *)&imageAllocProps;
+    imageAllocProps.pNext = (void *)&imageLinearPitchInfo;
+    ZE_CALL_NOCHECK(zeDeviceGetImageProperties, (ZeDevice, &imageProps));
+    return ReturnValue(imageLinearPitchInfo.maxSupportedPitch);
```
Comment (iclsrc) [inline unified-runtime/source/adapters/level_zero/device.cpp]: 🟡 **Important**: `UR_DEVICE_INFO_MAX_IMAGE_LINEAR_PITCH_EXP` is declared as `[size_t]` in `ur_api.h`, and the CUDA/HIP adapters return `size_t`. Here `maxSupportedPitch` is returned without any cast — if the underlying ZE field is `uint32_t` or `uint64_t`, the ABI contract could be violated silently. Please add an explicit `static_cast<size_t>()` to match the declared UR return type and make the intent clear, as is done for the CUDA adapter:
```suggestion
  case UR_DEVICE_INFO_MAX_IMAGE_LINEAR_PITCH_EXP: {
    ze_device_image_properties_t imageProps = {};
    imageProps.stype = ZE_STRUCTURE_TYPE_DEVICE_IMAGE_PROPERTIES;
    ze_device_pitched_alloc_exp_properties_t imageAllocProps = {};
    imageAllocProps.stype =
        ZE_STRUCTURE_TYPE_PITCHED_ALLOC_DEVICE_EXP_PROPERTIES;
    ze_pitched_alloc_2dimage_linear_pitch_exp_info_t imageLinearPitchInfo = {};
    imageLinearPitchInfo.stype = static_cast<ze_structure_type_t>(
        ZE_STRUCTURE_TYPE_PITCHED_ALLOC_2DIMAGE_LINEAR_PITCH_EXP_INFO);
    imageProps.pNext = (void *)&imageAllocProps;
    imageAllocProps.pNext = (void *)&imageLinearPitchInfo;
    ZE_CALL_NOCHECK(zeDeviceGetImageProperties, (ZeDevice, &imageProps));
    return ReturnValue(static_cast<size_t>(imageLinearPitchInfo.maxSupportedPitch));
  }
```

Comment (iclsrc) [inline unified-runtime/source/adapters/level_zero/device.cpp]: 🔵 **Suggestion**: Both new cases (`UR_DEVICE_INFO_IMAGE_PITCH_ALIGN_EXP` and `UR_DEVICE_INFO_MAX_IMAGE_LINEAR_PITCH_EXP`) duplicate the entire pNext chain construction (6 lines each). The existing `MAX_IMAGE_LINEAR_WIDTH_EXP` and `MAX_IMAGE_LINEAR_HEIGHT_EXP` cases also partially duplicate the same setup. Consider extracting a local lambda or helper that builds and queries the pNext chain once, keeping all four cases DRY and making future extensions (e.g. adding another field) a single-site change.

**Review by iclsrc — COMMENTED** [review]
This PR implements the two previously-unsupported L0 device queries (`UR_DEVICE_INFO_IMAGE_PITCH_ALIGN_EXP` and `UR_DEVICE_INFO_MAX_IMAGE_LINEAR_PITCH_EXP`) using a new ZE extension (`ze_pitched_alloc_2dimage_linear_pitch_exp_info_t`) and replaces the old backend-conditional test with a simpler all-backends test.

A few issues worth addressing before merge:

File: sycl/test-e2e/bindless_images/image_reqs_get_info.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    std::cout << "Max image linear row pitch = " << maxPitch << std::endl;
+    maxWidth = dev.get_info<sycl::ext::oneapi::experimental::info::device::
+                                max_image_linear_width>();
+    std::cout << "Max image linear width = " << maxWidth << std::endl;
+    maxheight = dev.get_info<sycl::ext::oneapi::experimental::info::device::
+                                 max_image_linear_height>();
+    std::cout << "Max image linear height = " << maxheight << std::endl;
+
+  } catch (sycl::exception e) {
+    std::cerr << "SYCL exception caught! : " << e.what() << "\n";
```
Comment (iclsrc) [inline sycl/test-e2e/bindless_images/image_reqs_get_info.cpp:40]: 🟡 **Important**: `sycl::exception` should be caught by const reference, not by value. Catching by value copies the exception object, which is wasteful and can slice derived types.
```suggestion
  } catch (const sycl::exception &e) {
```

File: sycl/test-e2e/bindless_images/image_reqs_get_info.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    return 1;
+  } catch (...) {
+    std::cerr << "Unknown exception caught!\n";
+    return 2;
+  }
+
```
Comment (iclsrc) [inline sycl/test-e2e/bindless_images/image_reqs_get_info.cpp:47]: 🟡 **Important**: The test has no pass/fail verdict output. The old `image_reqs_get_info.cpp` it replaces printed `"Test Passed!"`. More importantly, the new test does not validate that any of the returned values are non-zero on hardware that advertises the `aspect-ext_oneapi_bindless_images` capability. At minimum, add a `std::cout << "Test Passed!\n";` (or `return 0;`) marker so the LIT harness can confirm successful execution, and consider asserting the queried values are sane (e.g. `> 0`).

**Review by pbalcer — COMMENTED** [review]
lgtm, but please remove the code duplication.

**Comment by dyniols** [comment]
> lgtm, but please remove the code duplication.

I have removed the code duplication. Please take a look if you can @pbalcer. Thank you!

File: unified-runtime/source/adapters/level_zero/device.cpp (hunk 1 above)
Comment (dyniols) [reply unified-runtime/source/adapters/level_zero/device.cpp]: https://github.com/intel/llvm/pull/22087/commits/e2c5cc9fd954aada6b51742fa3632f8f0b059122

**Review by dm-vodopyanov — APPROVED** [review]

**Review by pbalcer — APPROVED** [review]

**Review by iclsrc — COMMENTED** [review]
Round 2 follow-up review. No Critical (🔴) or Important (🟡) issues found. The `ZeDevicePitchedAllocInfo` pNext chain is correctly established with stable addresses (ZeCache is non-movable via `std::once_flag`), the `UR_DEVICE_INFO_IMAGE_PITCH_ALIGN_EXP` return type matches the `uint32_t` contract, and the test scope change is properly guarded by the existing `REQUIRES: aspect-ext_oneapi_bindless_images` gate. LGTM.

**Comment by dyniols** [comment]
I think https://github.com/intel/llvm/actions/runs/26581525011/job/78328074197?pr=22087 failure `Adapters/level_zero/batch_test_copy_with_compute.cpp` is unrelated. 

Similar issue https://github.com/intel/llvm/issues/18552

**Comment by sys-ce-bb** [comment]
@intel/llvm-gatekeepers please consider merging
